filament/backend/src/vulkan/VulkanDriver.cpp

```diff
@@ -888,12 +895,52 @@ void VulkanDriver::beginRenderPass(Handle<HwRenderTarget> rth, const RenderPassP
         }
     }
 
-    const VkCommandBuffer cmdbuffer = mContext.commands->get().cmdbuffer;
     VulkanAttachment depth = rt->getSamples() == 1 ? rt->getDepth() : rt->getMsaaDepth();
+#ifdef VULKAN_DEBUG_REAL
+    if (depth.texture) {
+        depth.texture->print();
+    }
+#endif
 
-    VulkanDepthLayout initialDepthLayout = fromVkImageLayout(depth.getLayout());
-    VulkanDepthLayout renderPassDepthLayout = VulkanDepthLayout::ATTACHMENT;
-    VulkanDepthLayout finalDepthLayout = VulkanDepthLayout::ATTACHMENT;
+    bool samplingDepthAttachment = false;
+    const VkCommandBuffer cmdbuffer = mContext.commands->get().cmdbuffer;
+    for (uint8_t samplerGroupIdx = 0; samplerGroupIdx < Program::SAMPLER_BINDING_COUNT;
```

Comment:
clang tends to unroll loops that have a compile-time iteration count. It's not always what we want, so consider using `UTILS_NOUNROLL` if that's what you want.

Reply:
yes good suggestion done